### PR TITLE
Cleanup of merged pylab examples

### PR DESCRIPTION
**Pull request by QuLogic**
These are fixes I noticed in #8266 but never finished commenting. And anyway, since that was a merge-everything PR, trying to fill it with a lot of cleanup would be confusing.

Much of this is just trying to maintain some consistency within a file, e.g., if one example used OO and one example used pyplot, switch to just OO.

@choldgraf

### CODE REVIEW

File: examples/images_contours_and_fields/image_demo.py [hunk 1]
```diff
@@ -121,26 +122,21 @@
...
+    ax.grid(True)
 
 plt.show()
 
 
 ###############################################################################
 # You can specify whether images should be plotted with the array origin
-# x[0,0] in the upper left or upper right by using the origin parameter.
+# x[0,0] in the upper left or lower right by using the origin parameter.
 # You can also control the default be setting image.origin in your
```
Comment (choldgraf) [inline examples/images_contours_and_fields/image_demo.py]: I'm gonna mark typeos w/ `TYPEO` in the comment as a reminder to go back @ the end and fix the typos just before merging. (there's one in this line) "control the default be setting"

File: examples/pylab_examples/fancybox_demo.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def test4(ax):
     ax.add_patch(p_fancy)
 
     ax.text(0.1, 0.8,
-            ' boxstyle="round,pad=0.3"\n mutation\\_aspect=.5',
+            ' boxstyle="round,pad=0.3"\n mutation_aspect=.5',
```
Comment (choldgraf) [inline examples/pylab_examples/fancybox_demo.py]: I guess we don't have to worry about these getting rendered as math now? Or is that intended behavior now?

**Comment by choldgraf** [comment]
Just took a pass through it and I noticed very few problems (in comments) - it looks great to me and I think this is a good improvement towards making the docs more consistent!

File: examples/pylab_examples/fancybox_demo.py (hunk 2 above)
Comment (QuLogic) [inline examples/pylab_examples/fancybox_demo.py]: Only if wrapped in `$`, or maybe a `usetex=True` got removed at some point? They're definitely not needed right now though.

File: examples/pylab_examples/spy_demos.py [hunk 3]
```diff
@@ -7,15 +7,15 @@
...
 
 from matplotlib.pyplot import figure, show
-import numpy
+import numpy as np
 
 fig = figure()
 ax1 = fig.add_subplot(221)
 ax2 = fig.add_subplot(222)
 ax3 = fig.add_subplot(223)
 ax4 = fig.add_subplot(224)
```
Comment (QuLogic) [inline examples/pylab_examples/spy_demos.py]: Oh, I should probably change this one to `subplots`, too.

File: examples/pylab_examples/subplot_toolbar.py [hunk 4]
```diff
@@ -5,17 +5,17 @@
 
 """
 import matplotlib.pyplot as plt
-import numpy.random as rnd
+import numpy as np
 
 fig = plt.figure()
 plt.subplot(221)
```
Comment (QuLogic) [inline examples/pylab_examples/subplot_toolbar.py]: And one more `subplots` here as well.

**Comment by QuLogic** [comment]
Rebased to fix the typo and the two other `plt.subplots` calls I just noticed.

**Comment by QuLogic** [comment]
Just noticed that #8633 was not merged yet and it moves around a few of these files. I'd hold off on merging this one until that one's in, and I can rebase this one.

**Pull request merged**